syntax = "proto3";
package register;

service register {
    // CheckUser returns whether the username and password are correct
    rpc register(Request) returns (Response);
    rpc validateUsername(Request) returns (Response);
    rpc validatePassword(Request) returns (Response);
    rpc addClass(classRequest) return (classResponse);
  }

message Request {
    string userName = 1;
    string password = 2;
    string firstName = 3;
    string lastName= 4;
  }

message Response {
  bool success = 1;
}

message classRequest {
  string courseCode = 1;
  string section = 2;
}

message classResponse {
  string title = 1;
  string section = 2;
  string class_numbers = 3;
  string days = 4;
  string instructors = 5;
  string credit = 6;
}
